CONTRIBUTING.md

```diff
@@ -172,6 +172,12 @@ You can run end-to-end tests with:
 make e2e TEST_QUANTUM_PROCESSOR=<quantum processor ID>
 ```
 
```

Comment:
Should there be a note here describing how to _write_ these doctests? Maybe links to:

- https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html
- https://docs.python.org/3/library/doctest.html#option-flags

with a very brief primer so someone doesn't need to read all of those docs just to add a new function with working documentation. This change is great overall but I can see it becoming a stumbling block.

Reply:
I added a subsection with a quick guide on using doctest